arch/arm/src/cxd56xx/cxd56_sph.c

```diff
@@ -240,7 +240,7 @@ static inline int cxd56_sphdevinit(FAR const char *devname, int num)
 
   snprintf(fullpath, sizeof(fullpath), "/dev/%s%d", devname, num);
 
-  ret = register_driver(fullpath, &sph_fops, 0666, (FAR void *)priv);
+  ret = register_driver(fullpath, &sph_fops, 0000, (FAR void *)priv);
```

Comment:
should the device have read permission at least?

Reply:
Both `sph_fops.read` and `sph_fops.write` are `NULL` pointers. I do not think that read permission is needed. But maybe I'm lacking a vision of many things. `open` with mode `0` works perfectly, so this is not a limitation. I made this PR as draft and will initiate additional discussion over e-mail to get more input.

Reply:
We need consider ioctl too, actually Linux define ioctl with read/write in the encoding to consolidate the the permission check in one place:
https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/ioctl.h#L85-L91

Reply:
I tried to use ioctl with drivers that are registered with `0000` mode and that works like a charm. I noticed the mode issue while I was working on https://github.com/apache/incubator-nuttx/pull/5345
The driver is registered with `register_driver(devpath, &automount_fops, 0000, priv);` and test code does
```
  int fd = open("/var/mount/automount0", 0);
  if (fd > 0)
  {
...
        ret = ioctl(fd, FIOC_NOTIFY,
                  (unsigned long)((uintptr_t)&automountevents));
        if (ret < 0)
        {
          int errcode = errno;
          ferr("ERROR: ioctl(FIOC_NOTIFY) failed: %d\n",
               errcode);
        }
      else
        {
          /* Success */
        }
  }
```
The `ioctl` operation is always successful.
Maybe you can describe use case for your concern?

Reply:
I think currently these flags are not enforced on NuttX, but since we get multi-user support it will be necessary to use the right access mode flag.

Reply:
Yes. I will rework changes

Reply:
Done